pkg/loki/config_wrapper.go

```diff
@@ -111,17 +111,18 @@ func (c *ConfigWrapper) ApplyDynamicConfig() cfg.Source {
 // When using the ingester or common ring config, the loopback interface will be appended to the end of
 // the list of default interface names
 func applyDynamicRingConfigs(r, defaults *ConfigWrapper) {
-	if reflect.DeepEqual(r.Common.Ring, defaults.Common.Ring) {
-		// common ring not set, so use memberlist for all rings if set
-		if len(r.MemberlistKV.JoinMembers) > 0 {
-			applyMemberlistConfig(r)
-		} else {
-			// neither common ring nor memberlist set, use ingester ring configuration for all rings
-			useIngesterRingConfig(r, defaults)
-		}
-	} else {
-		// common ring is provided, use that for all rings
+	if !reflect.DeepEqual(r.Common.Ring, defaults.Common.Ring) {
+	        // common ring is provided, use that for all rings
 		useCommonRingConfig(r, defaults)
+		return
+	}
+
+	// common ring not set, so use memberlist for all rings if set
+	if len(r.MemberlistKV.JoinMembers) > 0 {
+		applyMemberlistConfig(r)
+	} else {
+		// neither common ring nor memberlist set, use ingester ring configuration for all rings
+		useIngesterRingConfig(r, defaults)
 	}
```

Comment:
I don't think we should apply the ingester configs to _any other ring_.

Reply:
If we make the decision, which I agree is more straight forward, we run the risk of 2.4 being a breaking release to people with existing configs, as it introduces a few new rings that are required